i18n-helpers/src/bin/mdbook-translate-helper.rs

```diff
@@ -0,0 +1,192 @@
+use std::env;
+use std::fs;
+use std::io;
+use std::io::BufRead;
+use std::io::Read;
+use std::io::Write;
+use::std::path::Path;
+use std::process::Command;
+use anyhow::Context;
+use anyhow::Error;
+use chrono;
+use walkdir::WalkDir;
+use zip::result::ZipError;
+use zip::write::SimpleFileOptions;
+use std::fs::File;
+
+fn build_translation(locale: String, dest_dir: String) -> Result<(), Error>{
```

Comment:
Functions very rarely need to take an owned `String` as an argument. Normally you want your functions to take `&str` instead.

The reason is that this gives the caller more flexibility: they can do both

```rust
let s = String::from("this is a heap-allocated owned string");
build_translation(&s);
```

and

```rust
build_translation("static non-heap allocated string");
```

When you take a `String` in your function signature, you force the caller to allocate memory for it.